zerver/forms.py

```diff
@@ -78,12 +91,34 @@ def __init__(self, *args, **kwargs):
     def clean_email(self):
         # type: () -> str
         data = self.cleaned_data['email']
-        if (get_unique_open_realm() or
+        if ((get_unique_open_realm() and not settings.OPEN_REALM_CREATION) or
```

Comment:
@timabbott Actually if we didn't check the the value of realm would be None and this would be treated as a realm registration because of this. It make sure that if REALM_CREATION is enabled all the registrations go through the domain specefic url. 
```python
if unique_open_realm is not None and not settings.OPEN_REALM_CREATION:
        realm = unique_open_realm
        domain = realm.domain
```
I will remove these and make it simpler by introducing another parameter as you suggested. Is it fine to change the Confirmation.object to include a parameter that differentiates realm and user registration?